✨  rename Frozen-Deps to Pinned-Dependencies

laurentsimon opened the pull request:
* **Please check if the PR fulfills these requirements**
- [] Tests for the changes have been added (for bug fixes / features)
- [X] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
feature


* **What is the current behavior?** (You can also link to an open issue here)
name is frozen-deps


* **What is the new behavior (if this is a feature change)?**
name is pinned-dependencies


* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
no


* **Other information**:
I also fix https://github.com/ossf/scorecard/issues/741 along the way.
Put all file-related functions into new file `fileutils.go`

inferno-chromium commented on checks/fileutils.go:
```diff
@@ -15,6 +15,7 @@
 package checks
```
add _ in filename, file_utils.go

laurentsimon replied:
done.

inferno-chromium commented on checks/fileutils.go:
```diff
@@ -106,3 +130,23 @@ func CheckFilesContent(shellPathFnPattern string,
+	s := ""
+	for scanner.Scan() {
+		line := strings.TrimSpace(scanner.Text())
+		s += "'" + line + "'\n"
```
s is unused ?

laurentsimon replied:
oops, that was debug code! removed.

inferno-chromium commented on checks/fileutils.go:
```diff
@@ -106,3 +130,23 @@ func CheckFilesContent(shellPathFnPattern string,
+
+// ContainsInstructions checks if the file contains instructions or not.
+// `comment` is the string or character that indicates a comment.
+func ContainsInstructions(content []byte, comment string) bool {
```
ContainsInstructions is slightly wrong name, since it is not contains but startsWith, maybe lineStartsWith ?
also, comment var name seems incorrect, since you are creating a generic function, so maybe rename to start or something like that ?

laurentsimon replied:
Maybe `ContainsCommands()`?

inferno-chromium commented on checks/file_utils.go:
```diff
@@ -106,3 +130,21 @@ func CheckFilesContent(shellPathFnPattern string,
 }
+
+// ContainsCommands checks if the file content contains commands or not.
+// `comment` is the string or character that indicates a comment.
```
change |comment| to |command| in this function and comments.

laurentsimon replied:
it's confusing. The function is used to see if the function contains commands. But it does so by checking if it contains only comments or empty lines. If we use comments instead of commands, the function should be called `ContainsOnlyCommentsOrEmptyLines` and I should change the return values false <-> true

laurentsimon replied:
changed to `CheckFileContainsCommands`. Can update in follow-up PR if necessary.

laurentsimon commented:
@naveensrinivasan how to I bypass the failing check? I've added empty files on purpose for unit tests.

naveensrinivasan commented:
> @naveensrinivasan how to I bypass the failing check? I've added empty files on purpose for unit tests.
> 
> @inferno-chromium


I don’t understand, so you want to bypass the integration test check?

Why not actually fix the issue? 

Can you please explain?

laurentsimon commented:
> I've added empty files on purpose for unit tests.

I've added empty files on purpose for unit tests (need to verify that our checks work properly on empty files). The check is failing because it wants every file to contain a copyright notice. 

Please advise. Thanks.
